File: cpp/example/src/SortIntByNum1Bits/sortIntByNum1Bits.cpp
#include "sortIntByNum1Bits.h"

bool sortIntByNum1Bits::comparison(const int x,const int y){
    if (x<y) return true;
    string sx = bitset<64>(x).to_string(); 
    string sy = bitset<64>(y).to_string(); 
    int num1x = count(sx.begin(),sx.end(),'1');
    int num1y = count(sy.begin(),sy.end(),'1');
    if (num1x<num1y) return true;
    return false;
};

vector<int> sortIntByNum1Bits::naive(vector<int>& arr){
    sort(arr.begin(),arr.end(),sortIntByNum1Bits::comparison);
    return arr;
}

File: cs/sample/Sample/SortIntByNum1Bits/SortIntByNum1Bits.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Sample.SortIntByNum1Bits{
    public class sortIntByNum1Bits{
        public static int Compare(int x,int y){
            string sx = Convert.ToString(x,2);
            string sy = Convert.ToString(y,2);
            int num1x = sx.Count(f => (f == '1'));
            int num1y = sy.Count(g => (g == '1'));
            if (x<y) return 0;
            if (num1x<num1y) return 0;
            return 1;
        }
        public static int[] naive(int[] arr){
            Array.Sort<int>(arr,sortIntByNum1Bits.Compare);
            return arr;
        }
    }
}
